scripts/exaqm_make_ics.sh

```diff
@@ -13,583 +13,13 @@ export pgm=aqm_make_ics
-# string "null".  This is equivalent to setting its value to
-#    !!python/none
-# in the base namelist file specified by FV3_NML_BASE_SUITE_FP or the
-# suite-specific yaml settings file specified by FV3_NML_YAML_CONFIG_FP.
-#
-# It turns out that setting the variable to an empty string also works
-# to remove it from the namelist!  Which is better to use??
-#
-settings="
-'config': {
- 'fix_dir_target_grid': ${FIXlam},
- 'mosaic_file_target_grid': ${FIXlam}/${CRES}${DOT_OR_USCORE}mosaic.halo$((10#${NH4})).nc,
- 'orog_dir_target_grid': ${FIXlam},
- 'orog_files_target_grid': ${CRES}${DOT_OR_USCORE}oro_data.tile${TILE_RGNL}.halo$((10#${NH4})).nc,
- 'vcoord_file_target_grid': ${FIXam}/global_hyblev.l65.txt,
- 'varmap_file': ${PARMdir}/ufs_utils/varmap_tables/${varmap_file},
- 'data_dir_input_grid': ${extrn_mdl_staging_dir},
- 'atm_files_input_grid': ${fn_atm},
- 'sfc_files_input_grid': ${fn_sfc},
- 'grib2_file_input_grid': \"${fn_grib2}\",
- 'cycle_mon': $((10#${mm})),
- 'cycle_day': $((10#${dd})),
- 'cycle_hour': $((10#${hh})),
- 'convert_atm': True,
- 'convert_sfc': True,
- 'convert_nst': ${convert_nst},
- 'regional': 1,
- 'halo_bndy': $((10#${NH4})),
- 'halo_blend': $((10#${HALO_BLEND})),
- 'input_type': ${input_type},
- 'external_model': ${external_model},
- 'tracers_input': ${tracers_input},
- 'tracers': ${tracers},
- 'nsoill_out': $((10#${nsoill_out})),
- 'geogrid_file_input_grid': ${geogrid_file_input_grid},
- 'vgtyp_from_climo': ${vgtyp_from_climo},
- 'sotyp_from_climo': ${sotyp_from_climo},
- 'vgfrc_from_climo': ${vgfrc_from_climo},
- 'minmax_vgfrc_from_climo': ${minmax_vgfrc_from_climo},
- 'lai_from_climo': ${lai_from_climo},
- 'tg3_from_soil': ${tg3_from_soil},
- 'thomp_mp_climo_file': ${thomp_mp_climo_file},
-}
-"
-#
-# Call the python script to create the namelist file.
-#
-nml_fn="fort.41"
-${USHaqm}/set_namelist.py -q -u "$settings" -o ${nml_fn}
-err=$?
-if [ $err -ne 0 ]; then
-  message_txt="FATAL ERROR Call to python script set_namelist.py to set the variables 
-in the namelist file read in by the ${exec_fn} executable failed. Parameters 
-passed to this script are:
-  Name of output namelist file:
-    nml_fn = \"${nml_fn}\"
-  Namelist settings specified on command line (these have highest precedence):
-    settings =
-$settings"
-    err_exit "${message_txt}"
-fi
-#
-#-----------------------------------------------------------------------
-#
-# Run chgres_cube.
-#
-#-----------------------------------------------------------------------
-#
-# NOTE:
-# Often when the chgres_cube.exe run fails, it still returns a zero
-# return code, so the failure isn't picked up the the logical OR (||)
-# below.  That should be fixed.  This might be due to the RUN_CMD_UTILS
-# command - maybe that is returning a zero exit code even though the
-# exit code of chgres_cube is nonzero.  A similar thing happens in the
-# forecast task.
-#
-startmsg
-eval ${RUN_CMD_UTILS} ${exec_fp} ${REDIRECT_OUT_ERR} >> $pgmout 2>errfile 
```

Comment:
The UW Driver YAML for chgres_cube.py now handles all of the information needed to define the component run environment, input files (data and configuration), and component execution.